pennylane/utils.py

```diff
@@ -323,6 +326,22 @@ def circuit2():
             + ",".join(string_reps)
         )
 
+    if qml.tape_mode_active():
+        for op in operation_list:
+            try:
+                qml.tape.QueuingContext.remove(op)
+            except KeyError:
+                pass
+
+        with qml.tape.QuantumTape() as tape:
+            for o in operation_list:
+                o.queue()
+                if o.inverse:
+                    o.inv()
+
+        tape.inv()
+        return tape
```

Comment:
This is too allow `qml.inv()` to work in tape mode. I was not previously re-written to support the new queuing system.

Here, we simply convert the list of operations into a sub/nested tape, remove the operations from the existing/parent tape, and invert the nested tape.

Reply:
isn't `qml.inv()` more of a transform (take a circuit and return the inverted circuit), rather than something we want to do "in-place" :thinking:

Reply:
Yep, agree, however this existing function was already an in-place function.

I heavily considered just deleting this function (not sure if anyone ever uses it, I had completely forgotten it existed 😆 ) but realised I could make it work with existing logic with these few added lines.

Reply:
But replacing this with a transform is definitely on the to-do list

Reply:
So this is now the only operation except from decompositions that introduces nesting, right? So if I use `inv` together with decompositions, the resulting tape can have nesting depth 3? And that is accounted for everywhere by the `tape.expand` function?